### ISSUE TITLE
[TRENDING CRASH] 0.1.9-Openforis LocalDataConsistencyException App Crash

### ISSUE THREAD

**Issue by anandwana001** [google/ground-android#2726]
```
 Fatal Exception: com.google.android.ground.persistence.local.LocalDataConsistencyException: Unknown jobId odVVCKPSrU02K31csXTf in location of interest wLVSAHW1I18pFWgL7Uwk
       at com.google.android.ground.persistence.local.room.converter.ConverterExtKt.toModelObject(ConverterExt.kt:153)
       at com.google.android.ground.persistence.local.room.stores.RoomLocationOfInterestStore.getLocationOfInterest(RoomLocationOfInterestStore.kt:65)
       at com.google.android.ground.persistence.local.room.stores.RoomLocationOfInterestStore$getLocationOfInterest$1.invokeSuspend(:15)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8810)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
        
```

**Comment by gino-m** [comment]
The surveys and jobs reported in logs exist in remote, so this is also likely fixed by #2740. 

@jabramowitz5 Can you please have field crews remove surveys from their device and reopen, avoiding to tap multiple times on the survey card while loading? Let's check back on this once everyone has done that.

**Comment by gino-m** [comment]
I just observed this locally when debugging #2377. Setting to P0 for urgent attention @anandwana001

**Comment by gino-m** [comment]
Possible culprit:

https://github.com/google/ground-android/blob/df3fbaf8fe1b21ad444c7f423b9797f9490244a7/ground/src/main/java/com/google/android/ground/persistence/local/room/stores/RoomSurveyStore.kt#L66-L70

When the survey is updated on disk, all  jobs are deleted and recreated. This operation is not isolated in a transaction, which can lead to a race condition with the data sync process. We should instead wrap the operation in a transaction and/or stop deleting all jobs and update in place instead.

**Comment by gino-m** [comment]
Recommended steps:

- [ ] Update survey metadata in a single transaction.
- [ ] Mutations are no longer flagged FAILED on error - fix this.
- [ ] Periodically check for "stuck" mutations and resync. To do this we way want to update the `LocalMutationSyncWorker` to look at pending mutations for all LOIs each time its run instead of per-LOI. We should just be sure to group by LOIs and catch all errors between iterations to ensure a failure of one LOI-submission doesn't prevent others from syncing.
- [ ] Stop using "delete+insert" instead of "update" throughout and enable cascading on all foreign keys to guarantee consistency of local db and prevent accidental deletion of data.

**Comment by gino-m** [comment]
> Ensure mutations are marked FAILED when there's an error

We do set mutations to FAILED when they fail inside `processMutations()`, but not in the outermost try/catch. Some work is needed here to rationalize the entire flow (#2840).

**Comment by gino-m** [comment]
There are too many instances for this to have been caused by a race condition. Reopening just to try to understand better.

**Comment by gino-m** [comment]
@anandwana001 Can you help with this.

**Comment by gino-m** [comment]
> Stop using "delete+insert" instead of "update" throughout and enable cascading on all foreign keys to guarantee consistency of local db and prevent accidental deletion of data.

Moved this code health task into #2840. Closing this under the assumption it is closed by #2835. Please reopen if the issue recurs.
